### PR TITLE
Zijian/implement find for status

### LINKED ISSUE #17: As a user, I can find tasks by due date.

### LINKED ISSUE #146: Implement find by status

### PR DESCRIPTION
**Pull request by BobbyZhouZijian**
Find now works for status.

Must search for the exact word of the status. E.g. "incomplete" will find all incomplete tasks, but "inc" will not match any.

Have not refactored the docs yet. Waiting for some kind guy to fix it. I'm too sleepy now and need to rest.

Fix #146 
Fix #17

### CODE REVIEW

**Comment by MarcTzh** [comment]
Hi Zijian, I conducted some boundary value analysis and found that it works as intended for the positive test cases but when i put in find status: it does not throw any error, it says 0 tasks listed, following equivalence partitioning i shall stop further tests

**Comment by MarcTzh** [comment]
A few problems I found: 
`find status: XXX` where XXX is an invalid input, returns 0 tasks found as well

Error message displayed should state that status can only be of 2 values: complete and incomplete.

`find date:01-01-202` is also accepted and the date 01-01-2020 matches it, should this be considered an incorrect input

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -288,7 +288,7 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 **MSS**
 
     1. User requests to add a specific task to the list.
-
+
```
Comment (MarcTzh) [inline docs/DeveloperGuide.md]: do not add all the white spaces

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -391,16 +391,28 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+- 1b. The search phrase is in invalid format or includes invalid characters.
+
+    - 1b1. PlaNus shows an error message.
+
+        Use case ends.
+
 - 2a. The list is empty.
 
-  Use case ends.
+    ​	Use case ends.
```
Comment (MarcTzh) [inline docs/DeveloperGuide.md]: no space needed here

File: docs/UserGuide.md [hunk 3]
```diff
@@ -112,12 +112,23 @@ Examples:
...
 
 Format: `find ATTRIBUTE_1:SEARCH_PHRASE ATTRIBUTE_2:SEARCH_PHRASE ...`
 
-Available attributes in v1.3 include:
 
+
+If different attributes are supplied, tasks that match all attributes will be displayed.
+If multiple search phrases of the same attribute is supplied, tasks that match any of the 
+search phrase will be displayed.
+
```
Comment (MarcTzh) [inline docs/UserGuide.md:122]: why are there so many empty lines

File: docs/UserGuide.md [hunk 4]
```diff
@@ -112,12 +112,23 @@ Examples:
...
+
+
+Search phrases supplied must be valid input.
+
+
```
Comment (MarcTzh) [inline docs/UserGuide.md]: ^

File: docs/UserGuide.md [hunk 5]
```diff
@@ -112,12 +112,23 @@ Examples:
...
+
+Available attributes in v1.3 include:
 * `title:` find all tasks with the search phrase in the given title
 * `desc:` find all tasks with the search phrase in the given description
 * `type:` find all tasks of the given type
```
Comment (MarcTzh) [inline docs/UserGuide.md:128]: what about status

File: src/main/java/seedu/address/model/task/Status.java [hunk 6]
```diff
@@ -10,7 +10,7 @@
 
 
     public static final String MESSAGE_CONSTRAINTS =
-            "The value is from the enum class State";
+            "Values must be either 'complete' or 'incomplete'";
```
Comment (MarcTzh) [inline src/main/java/seedu/address/model/task/Status.java]: Value of status* - this makes it easier for the user to determine what is wrong with their input if there are multiple attributes in a find command

**Review by MarcTzh — COMMENTED** [review]
Minor changes

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -398,9 +398,21 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+- 1b. The search phrase is in invalid format or includes invalid characters.
+
+    - 1b1. PlaNus shows an error message.
+
+    Use case ends.
+
 - 2a. The list is empty.
 
-  Use case ends.
+    ​Use case ends.
```
Comment (MarcTzh) [inline docs/DeveloperGuide.md]: Remove space

**Review by MarcTzh — COMMENTED** [review]
one minor change, LGTM

File: docs/DeveloperGuide.md (hunk 7 above)
Comment (BobbyZhouZijian) [reply docs/DeveloperGuide.md]: resolved

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -398,9 +398,21 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+- 1b. The search phrase is in invalid format or includes invalid characters.
+
+    - 1b1. PlaNus shows an error message.
+
+    Use case ends.
+
 - 2a. The list is empty.
 
-  Use case ends.
+   ​Use case ends.
```
Comment (MarcTzh) [inline docs/DeveloperGuide.md]: remove this space

**Review by MarcTzh — APPROVED** [review]

File: src/main/java/seedu/address/model/task/TaskContainsKeywordsPredicate.java [hunk 9]
```diff
@@ -26,20 +26,26 @@ public void setKeyword(Prefix attribute, String keyword) {
...
     }
 
     @Override
     public boolean test(Task task) {
+        boolean matched = false;
         for (Map.Entry<Prefix, List<String>> entry : keywords.entrySet()) {
             Prefix prefix = entry.getKey();
             List<String> words = entry.getValue();
-            if (isMatched(prefix.getPrefix(), words, task)) {
-                return true;
```
Comment (dearvae) [inline src/main/java/seedu/address/model/task/TaskContainsKeywordsPredicate.java:37]: I feel the original code is correct?
